### PR TITLE
Replace configuration with ISecretService for secrets

### PR DESCRIPTION
**Pull request by misha130**
### Pull Request summary

Adds a ISecretService to get secrets when ever is needed instead of configuration

### Link to Project Card/Issue

https://github.com/codidact/core/issues/56

### Pull Request Tasklist

- [x] Pull request references an issue
- [x] Branch follows contributing guidelines format
- [x] Pull request targets the `develop` branch

### CODE REVIEW

File: tests/Infrastructure.IntegrationTests/Secrets/DevelopmentSecretsServiceTest.cs
Comment (asynts) [inline tests/Infrastructure.IntegrationTests/Secrets/DevelopmentSecretsServiceTest.cs]: The formatting is a bit weird here.

**Review by asynts — APPROVED** [review]

File: tests/Infrastructure.IntegrationTests/Secrets/DevelopmentSecretsServiceTest.cs
Comment (misha130) [reply tests/Infrastructure.IntegrationTests/Secrets/DevelopmentSecretsServiceTest.cs]: Need to remove extra newline and add extra line between prop and method. Any chance you can suggest?

File: tests/Infrastructure.IntegrationTests/Secrets/DevelopmentSecretsServiceTest.cs
Comment (asynts) [inline tests/Infrastructure.IntegrationTests/Secrets/DevelopmentSecretsServiceTest.cs]:
```suggestion
        private readonly Dictionary<string, string> InMemoryConfiguration =
            new Dictionary<string, string>
            {
                { "DefaultConnection", "LocalDb" },
            };

        public DevelopmentSecretsServiceTest()
        {
            var config = new ConfigurationBuilder()
                .AddInMemoryCollection(InMemoryConfiguration).Build();
```

File: src/Infrastructure/DependencyInjection.cs
Comment (poke) [inline src/Infrastructure/DependencyInjection.cs]: `async void`? This won’t work. `AddDbContext(…)` configures options, and options configuration is synchronous. If you have an async process here to retrieve the configuration from, then this won’t be able to synchronously configure the database context.

File: src/WebUI/Options/OpenIdConnectPostConfigureOptions.cs
Comment (poke) [inline src/WebUI/Options/OpenIdConnectPostConfigureOptions.cs]: `PostConfigure` is called synchronously, so when the secrets service implementation actually runs asynchronously (which the `DevelopmentSecretsService` does not do), then this will not work.

File: src/WebUI/Startup.cs
Comment (poke) [inline src/WebUI/Startup.cs]: You could also consider using the `AddOptions()` pattern here to avoid having to implement a post-configure action for a single line:

```
services.AddOptions<OpenIdConnectOptions>("oidc")
    .Configure<ISecretsService>((options, secretsService) =>
    {
        // set options with secrets service
    });
```

File: src/Infrastructure/DependencyInjection.cs
Comment (misha130) [reply src/Infrastructure/DependencyInjection.cs]: Switched to `.GetAwaiter().GetResult()` to make it sync. Testing async with sync was not a good idea.

File: src/WebUI/Options/OpenIdConnectPostConfigureOptions.cs
Comment (misha130) [reply src/WebUI/Options/OpenIdConnectPostConfigureOptions.cs]: Switched to .GetAwaiter().GetResult() to make it sync.

File: src/WebUI/Startup.cs
Comment (misha130) [reply src/WebUI/Startup.cs]: Done that

**Comment by cellio** [comment]
Are we waiting for more changes or for a review?

**Comment by asynts** [comment]
In your last commit you added a ton of files in `.idea`. I do not think that should be included in the repository.

Can you rebase that?

File: src/WebUI/Startup.cs
Comment (asynts) [inline src/WebUI/Startup.cs:66]:
```suggestion
                .Configure<ISecretsService>((options, secretsService) =>
                {
                    options.ClientSecret = secretsService.Get("Identity:ClientSecret").GetAwaiter().GetResult();
                }); 
```

File: .idea/.gitignore
Comment (asynts) [inline .idea/.gitignore]: I do not think that this directory should be added to source control.

Everybody uses their own IDE with their own options, some people use Visual Studio, some use Visual Studio Code and others use Vim.

Comment (misha130) [reply .idea/.gitignore]: An accident. It didn't show up in the list of changes when I committed in Visual Studio. Rebased.

**Comment by ArtOfCode-** [comment]
Conflicts need resolving. After that, are we ready for merge?

**Comment by misha130** [comment]
Had to include the instructions for setting up the secrets in linux/windows. Please review

**Review by ArtOfCode- — APPROVED** [review]

**Pull request merged**
